src/bson.rs

```diff
@@ -361,11 +367,11 @@ impl From<Bson> for Value {
                     "i": inc
                 })
             }
-            Bson::Binary(t, ref v) => {
-                let tval: u8 = From::from(t);
+            Bson::Binary(Binary { subtype, ref bytes }) => {
+                let tval: u8 = From::from(subtype);
                 json!({
```

Comment:
This is also the legacy extJSON. This should be `{ "$binary": {"base64": <base64-encoded (with padding as =) payload as a string>, "subType": <BSON binary type as a one- or two-character hex string>}}`